### PR TITLE
Add tests for bad parquet files

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/6261

# Rationale for this change
 
https://github.com/apache/parquet-testing has a a bunch of negative tests with invalid parquet data of various forms

We should ensure the rust reader properly handles these cases (errors / panics etc), for example

# What changes are included in this PR?

1. update the `parquet-testing` pin to https://github.com/apache/parquet-testing/pull/58
2. Add test that reads the parquet files

# Are there any user-facing changes?
No this PR is all tests

### CODE REVIEW

File: parquet/tests/arrow_reader/bad_data.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+fn test_arrow_rs_gh_6229_dict_header() {
+    let err = read_file("ARROW-RS-GH-6229-DICTHEADER.parquet").unwrap_err();
+    assert_eq!(
+        err.to_string(),
+        "External: Parquet argument error: EOF: eof decoding byte array"
+    );
+}
+
+#[test]
+fn test_arrow_rs_gh_6229_dict_levels() {
```
Comment (alamb) [inline parquet/tests/arrow_reader/bad_data.rs:102]: Prior to https://github.com/apache/arrow-rs/pull/6232 this test generated an infinite loop

File: parquet/tests/arrow_reader/bad_data.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+use std::collections::HashSet;
+use std::path::PathBuf;
+
+static KNOWN_FILES: &[&str] = &[
+    "PARQUET-1481.parquet",
+    "ARROW-GH-41317.parquet",
+    "ARROW-GH-41321.parquet",
+    "ARROW-RS-GH-6229-DICTHEADER.parquet",
+    "ARROW-RS-GH-6229-LEVELS.parquet",
+    "README.md",
```
Comment (mapleFU) [inline parquet/tests/arrow_reader/bad_data.rs:32]: aha!

**Comment by mapleFU** [comment]
LGTM!

**Comment by alamb** [comment]
Thank you for the review @mapleFU

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Pull request merged**